File: submissions/fizz-buzz.py
# https://leetcode.com/problems/fizz-buzz

class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        answer = [''] * n
        for i in range(1, n+1):
            cur = ''
            if i % 3 == 0:
                cur += 'Fizz'
            if i % 5 == 0:
                cur += 'Buzz'
            if len(cur) == 0:
                cur = str(i)
            answer[i-1] = cur
        return answer
        

File: submissions/fizz-buzz.go
/*
 * @lc app=leetcode id=412 lang=golang
 *
 * [412] Fizz Buzz
 */

// @lc code=start
import (
	"strconv"
)

func fizzBuzz(n int) []string {
	res := make([]string, n+1)

	for i := 1; i <= n; i++ {
		if i%3 == 0 {
			res[i] += "Fizz"
		}
		if i%5 == 0 {
			res[i] += "Buzz"
		}
		if i%3 != 0 && i%5 != 0 {
			res[i] += strconv.Itoa(i)
		}
	}

	return res[1:]
}

// @lc code=end

